lib/iris/fileformats/nimrod_load_rules.py

```diff
@@ -317,8 +700,175 @@ def add_attr(name):
+    Adds a coord relating to probability meta-data from the header to the
+    cube if appropriate.
+    Returns True if this is a blended multi-member field
+    """
+    probtype_lookup = {
+        1: {
+            "var_name": "threshold",
+            "attributes": {"relative_to_threshold": "above"},
+        },
+        2: {
+            "var_name": "threshold",
+            "attributes": {"relative_to_threshold": "below"},
+        },
+        3: {"long_name": "percentile", "units": "1"},
+        4: {
+            "var_name": "threshold",
+            "attributes": {"relative_to_threshold": "equal"},
+        },
+    }
+    probmethod_lookup = {
+        1: "AOT (Any One Time)",
+        2: "ST (Some Time)",
+        4: "AT (All Time)",
+        8: "AOL (Any One Location)",
+        16: "SW (Some Where)",
+    }
+    is_multi_member_field = False
+    coord_keys = probtype_lookup.get(field.threshold_type, {})
+    if coord_keys:
+        is_multi_member_field = True
+    coord_keys.update(known_threshold_coord(field))
+    if not coord_keys.get("units", None):
+        coord_keys["units"] = DEFAULT_UNITS.get(field.field_code, None)
+    coord_val = None
+    if field.threshold_value_alt > -32766.0:
+        coord_val = field.threshold_value_alt
+    elif field.threshold_value > -32766.0:
+        coord_val = field.threshold_value
+    if field.title.find("pc") > 0:
+        try:
+            coord_val = [
+                int(x.strip("pc"))
+                for x in field.title.split(" ")
+                if x.find("pc") > 0
+            ][0]
+        except IndexError:
+            pass
+    if coord_val is not None:
+        if field.threshold_fuzziness > -32766.0:
+            bounds = [
+                coord_val * field.threshold_fuzziness,
+                coord_val * (2.0 - field.threshold_fuzziness),
+            ]
+            bounds = np.array(bounds, dtype=np.float32)
+        else:
+            bounds = None
+        if coord_keys.get("units", None) == "oktas":
+            coord_keys["units"] = "1"
+            coord_val /= 8.0
+            if bounds is not None:
+                bounds /= 8.0
+        new_coord = iris.coords.AuxCoord(
+            np.array(coord_val, dtype=np.float32), bounds=bounds, **coord_keys
+        )
+        cube.add_aux_coord(new_coord)
+        if field.threshold_type == 3:
+            pass
+        else:
+            if is_multi_member_field:
+                cube.units = "1"
+                cube.rename(f"probability_of_{cube.name()}")
+
+    if field.probability_method > 0:
+        probability_attributes = []
+        num = field.probability_method
+        for key in sorted(probmethod_lookup.keys(), reverse=True):
+            if num >= key:
+                probability_attributes.append(probmethod_lookup[key])
+                num = num - key
+        cube.attributes["Probability methods"] = probability_attributes
+    if field.member_count == 1:
+        is_multi_member_field = False
```

Comment:
I think the logic here is a bit tricky ?
We started out above be setting is_multi_member_field=True if we spot a recognised field.threshold_type
This was then used to trigger the field name prefix, unless it's type 3 (line 833).
But now we cancel it, for the return value, if we only had 1 member field.

In the caller (`run()`), that switches whether `ensemble_member()` gets called.
I'm not quite clear on the requirements, but it does seem to "mean" two slightly different things within this routine + outside it (the return value), so at least a comment may be useful.

Reply:
> I think the logic here is a bit tricky ?
> We started out above by setting `is_multi_member_field`=True if we spot a recognised field.threshold_type
> This was then used to trigger the field name prefix, unless it's type 3 (line 833).
> But now we cancel it, for the return value, if we only had 1 member field.
> 
> In the caller (`run()`), that switches whether `ensemble_member()` gets called.
> I'm not quite clear on the requirements, but it does seem to "mean" two slightly different things within this routine + outside it (the return value), so at least a comment may be useful.

Reply:
I've split this logic into two separate variables to avoid confusion.

Reply:
This looks okay to me.
I see that you completely removed the linkage whereby this routine determines whether 'ensemble_member' gets called, so I assume that makes sense ...